utils/analyticsmetrics_test.go

```diff
@@ -0,0 +1,160 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	coretests "github.com/jfrog/jfrog-cli-core/v2/common/tests"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-client-go/xray/services"
+	xscservices "github.com/jfrog/jfrog-client-go/xsc/services"
+	"github.com/owenrumney/go-sarif/v2/sarif"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	testMsi                             = "27e175b8-e525-11ee-842b-7aa2c69b8f1f"
+	lowerAnalyticsMetricsMinXscVersion  = "1.6.0"
+	higherAnalyticsMetricsMinXscVersion = "1.10.0"
+)
+
+func TestCalcShouldReportEvents(t *testing.T) {
+	// Save original environment information.
+	originalJfMsi := os.Getenv(jfMsiEnvVariable)
+	defer func() {
+		err := os.Setenv(jfMsiEnvVariable, originalJfMsi)
+		assert.NoError(t, err)
+	}()
+	originalReportUsage := os.Getenv(coreutils.ReportUsage)
+	defer func() {
+		err := os.Setenv(coreutils.ReportUsage, originalReportUsage)
+		assert.NoError(t, err)
+	}()
+
+	err := os.Setenv(coreutils.ReportUsage, "")
+	assert.NoError(t, err)
+	err = os.Setenv(jfMsiEnvVariable, "")
+	assert.NoError(t, err)
+
+	// Minimum Xsc version.
+	mockServer, serverDetails := xscServer(t, AnalyticsMetricsMinXscVersion)
+	defer mockServer.Close()
+	am := NewAnalyticsMetricsService(serverDetails)
+	assert.True(t, am.calcShouldReportEvents())
+
+	// Lower Xsc version.
+	mockServer, serverDetails = xscServer(t, lowerAnalyticsMetricsMinXscVersion)
+	defer mockServer.Close()
+	am = NewAnalyticsMetricsService(serverDetails)
+	assert.False(t, am.calcShouldReportEvents())
+
+	// Higher Xsc version.
+	mockServer, serverDetails = xscServer(t, higherAnalyticsMetricsMinXscVersion)
+	defer mockServer.Close()
+	am = NewAnalyticsMetricsService(serverDetails)
+	assert.True(t, am.calcShouldReportEvents())
+
+	// JF_MSI was already provided.
+	err = os.Setenv(jfMsiEnvVariable, "msi")
+	assert.NoError(t, err)
+	assert.False(t, am.calcShouldReportEvents())
+
+	// JFROG_CLI_REPORT_USAGE is false.
+	err = os.Setenv(jfMsiEnvVariable, "")
+	assert.NoError(t, err)
+	err = os.Setenv(coreutils.ReportUsage, "false")
+	assert.NoError(t, err)
+	assert.False(t, am.calcShouldReportEvents())
```

Comment:
Define and use Test cases:

testCases := []struct {
    TestName string
    Version string
    expectedResult bool
}{
    ...
}

Reply:
this structure is not suitable for this specific test.